index.d.ts

```diff
@@ -0,0 +1,249 @@
+declare module 'ember-concurrency' {
+  import Task from 'ember-concurrency/task';
+
+  type TaskDecorator = PropertyDecorator & {
+    drop(): TaskDecorator;
+    keepLatest(): TaskDecorator;
+    restartable(): TaskDecorator;
+    maxConcurrency(num: number): TaskDecorator;
+    enqueue(): TaskDecorator;
+    withTestWaiter(): PropertyDecorator;
+  };
+
+  export function task(
+    generator: Function
+  ): PropertyDecorator & {
+    drop(): TaskDecorator;
+    keepLatest(): TaskDecorator;
+    restartable(): TaskDecorator;
+    maxConcurrency(num: number): TaskDecorator;
+    enqueue(): TaskDecorator;
+    withTestWaiter(): PropertyDecorator;
+  };
+  export function timeout(wait: number): Promise<void>;
+
+  export function didCancel(error: Error): boolean;
+}
+
+declare module 'ember-concurrency/task' {
+  import TaskInstance from 'ember-concurrency/task-instance';
+
+  export enum TaskState {
+    RUNNING = 'running',
+    QUEUED = 'queued',
+    IDLE = 'idle',
+  }
+
+  /**
+   * The `Task` object lives on a host Ember object (e.g.
+   * a Component, Route, or Controller). You call the
+   * `.perform()` method on this object
+   * to create run individual `TaskInstance`s,
+   * and at any point, you can call the `.cancelAll()`
+   * method on this object to cancel all running or enqueued
+   * `TaskInstance`s.
+   */
+  export default class Task<PerformArgs extends any[] = any[], PerformReturn = void> {
+    /**
+     * `true` if any current task instances are running.
+     */
+    readonly isRunning: boolean;
+
+    /**
+     * `true` if any future task instances are queued.
+     */
+    readonly isQueued: boolean;
+
+    /**
+     * `true` if the task is not in the running or queued state.
+     */
+    readonly isIdle: boolean;
+
+    /**
+     * The current state of the task: `"running"`, `"queued"` or `"idle"`.
+     */
+    readonly state: TaskState;
+
+    /**
+     * The most recently started task instance.
+     */
+    readonly last: TaskInstance<PerformArgs, PerformReturn>;
```

Comment:
this (and the other `last*` properties) can also be `null`. Do these need to be denoted as nullable?

Reply:
oh! yes, you're absolutely right. (though is it null or undefined?)